modules/offers/client/components/OfferLocationPresentational.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import L from 'leaflet';
+
+import { Map, TileLayer, Circle, Marker } from 'react-leaflet';
+
+export default function OfferLocationPresentational({ zoom, location, onChangeZoom, marker='', windowWidth }) {
+
+  return (
+    <Map
+      className="offer-location"
+      attributionControl={false}
+      zoom={zoom}
+      center={location}
+      onZoom={({ target: { _zoom: zoom } }) => onChangeZoom(zoom)}
+      scrollWheelZoom={false}
+    >
+      <TileLayer
+        attribution='&amp;copy <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+      />
+      {/* @TODO Circle and Marker will need to be reusable when we migrate the /search to React */}
+      {(zoom >= 12) ?
+        <Circle
+          center={location}
+          radius={500}
+          weight={2}
+          color="#989898"
+          fillColor="#b1b1b1"
+          fillOpacity={0.5}
+          clickable={false}
+          interactive={false}
+          layer="locationPath"
+        /> :
+        <Marker
+          position={location}
+          layer="locationMarker"
+          clickable={false}
+          focus={false}
+          icon={getIcon(marker, windowWidth)}
+        />
+      }
+    </Map>
+  );
+}
+
+/**
+ * Get url of marker icon
+ */
+function getMarkerUrl(marker) {
+  const markerName = ['marker', 'icon', ...((marker) ? [marker] : [])].join('-');
+  return `/img/map/${markerName}.svg`;
+};
+
+/**
+ * Size of the map icon in pixels (bigger for smaller screens)
+ * @param {Number} windowWidth - current width of window
+ * @returns {Number} - icon size
+ */
+function getIconSize(windowWidth) {
+  return (windowWidth < 768) ? 30 : 20;
```

Comment:
This is okay now in this PR, but just noting that I'm looking forward to using [`window.matchMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) or even something like this helper:

https://github.com/WordPress/gutenberg/tree/0f18024e477bafec475ec3785bb2209562a961c4/packages/viewport#readme

(There are probably more generic ones than the WordPress one)

Reply:
added TODO in 0ec4e507a9e258e0cad0b481a421c7607ef6a07f